src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/ConnectionHandlerContext.cs

```diff
@@ -58,12 +59,24 @@ internal class ConnectionHandlerContext : HandlerRequest, ICloneable
         /// Whether the connection is capable of MARS
         /// This is negotiated after pre-login.
         /// </summary>
-        public bool MarsCapable { get; internal set; }
+        public bool IsMarsCapable { get; internal set; }
 
         /// <summary>
         /// Indicates if fed auth needed for this connection.
         /// </summary>
-        public bool FedAuthRequired { get; internal set; }
+        public bool IsFedAuthRequired { get; internal set; }
+
+        /// <summary>
+        /// Get if data classification is enabled by the server.
+        /// </summary>
+        public bool IsDataClassificationEnabled =>
```

Comment:
DataClassification enabled is 2 things
1. That the feature extension is acknowledged,
2. The FE is acknowledged with the valid version.
This change only brings in the version check. i.e. point 2. Point 1 needs to be accounted for as well, unless you mean that invalid version is also set for servers which dont acknowledge the FE. 

Typically the 2 checks above must be performed. The client will send the FE bit with the data signifying the highest Version it supports. Hence the server will always respond with the Min of the highest version it supports and the client's requested version.